### PR TITLE
Corrected chinese braille table names

### LINKED ISSUE #13674: Wrong display name for Chinese Braille output and input tables
### Steps to reproduce:
1. Switch the display language of NVDA to English;
2. Go to the Braille settings panel;
3. Navigate the output table and input table combo box.
### Actual behavior:
You will see the following items:
- Chinese (China, Mandarin) grade 1
- Chinese (China, Mandarin) grade 2

### Expected behavior:
The display name here is incorrect, according to the definition in the Chinese National Standard [GB/T 15720-2008][1], sections 2.13 and 2.14:
- "Chinese (China, Mandarin) grade 1" should be changed to "Chinese (China, Mandarin) Current Braille System"
- "Chinese (China, Mandarin) grade 2" should be changed to "Chinese (China, Mandarin) Double-phonic Braille System"

I will open a PR to correct this.

Due to copyright reasons, the text in the above "GB/T 15720-2008" link cannot be read by visually impaired users, but it can be previewed in image form.
#### NVDA installed/portable/running from source:
installed, portable, running from source:
#### NVDA version:
All
#### Windows version:
n/a
#### Name and version of other software in use when reproducing the issue:
n/a
#### Other information about your system:
n/a
#### Does the issue still occur after restarting your computer?
n/a
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
n/a
#### If NVDA add-ons are disabled, is your problem still occurring?
n/a
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
n/a

[1]: http://www.gb688.cn/bzgk/gb/newGbInfo?hcno=484E4150A1F62E81B0553422E6A8A292

### PR DESCRIPTION
**Pull request by cary-rowen**
### Link to issue number:
Close #13674 
### Summary of the issue:
The following entries exist in the Output Table and Input Table combo boxe in the NVDA Braille Settings panel:
- Chinese (China, Mandarin) grade 1
- Chinese (China, Mandarin) grade 2

According to the definition in the Chinese National Standard [GB/T 15720-2008][1], sections 2.13 and 2.14:
- "Chinese (China, Mandarin) grade 1" should be changed to "Chinese (China) Current Braille System"
- "Chinese (China, Mandarin) grade 2" should be changed to "Chinese (China) Double-phonic Braille System"

Due to copyright reasons, the text in the above "GB/T 15720-2008" link cannot be read by visually impaired users, but it can be previewed in image form.

### Description of how this pull request fixes the issue:
I modified the corresponding string in source/brailleTables.py.
### Testing strategy:
manual testing
### Known issues with pull request:
None
### Change log entries:
New features

Changes

Corrected chinese braille table names:
  - "Chinese (China, Mandarin) grade 1" changed to "Chinese (China) Current Braille System"
  - "Chinese (China, Mandarin) grade 2" changed to "Chinese (China) Double-phonic Braille System"

Bug fixes

For Developers

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

[1]: http://www.gb688.cn/bzgk/gb/newGbInfo?hcno=484E4150A1F62E81B0553422E6A8A292

### CODE REVIEW

**Comment by cary-rowen** [comment]
Hi @seanbudd, this is a small fix, can it be merged?

File: source/brailleTables.py [hunk 1]
```diff
@@ -526,10 +526,10 @@ def listTables():
 addTable("xh-za-g2.ctb", _("Xhosa grade 2"), contracted=True)
 # Translators: The name of a braille table displayed in the
 # braille settings dialog.
-addTable("zhcn-g1.ctb", _("Chinese (China, Mandarin) grade 1"))
+addTable("zhcn-g1.ctb", _("Chinese (China, Mandarin) Current Braille System"))
 # Translators: The name of a braille table displayed in the
 # braille settings dialog.
-addTable("zhcn-g2.ctb", _("Chinese (China, Mandarin) grade 2"), contracted=True)
+addTable("zhcn-g2.ctb", _("Chinese (China, Mandarin) Double-phonic Braille System"), contracted=True)
```
Comment (seanbudd) [inline source/brailleTables.py:532]:
```suggestion
# This should be translated to 漢語雙拼盲文 in Mandarin.
addTable("zhcn-g2.ctb", _("Chinese (China, Mandarin) Double-phonic Braille System"), contracted=True)
```

File: source/brailleTables.py [hunk 2]
```diff
@@ -526,10 +526,10 @@ def listTables():
 addTable("xh-za-g2.ctb", _("Xhosa grade 2"), contracted=True)
 # Translators: The name of a braille table displayed in the
 # braille settings dialog.
-addTable("zhcn-g1.ctb", _("Chinese (China, Mandarin) grade 1"))
+addTable("zhcn-g1.ctb", _("Chinese (China, Mandarin) Current Braille System"))
```
Comment (seanbudd) [inline source/brailleTables.py:529]:
```suggestion
# This should be translated to 现行盲文 in Mandarin.
addTable("zhcn-g1.ctb", _("Chinese (China, Mandarin) Current Braille System"))
```

**Review by seanbudd — COMMENTED** [review]
If you are expecting the translators to translate this to a specific string, it's worth adding a message in the translator comments. Use my comments below as an example.

**Comment by cary-rowen** [comment]
HI @seanbudd 
I can't seem to successfully add a translator's note with Chinese characters, otherwise Gettext will throw an error.
So, even though I applied your suggested changes, translators using Poedit cannot see the comments.

**Comment by CyrilleB79** [comment]
You may give a try be defining the coding as first line of your file:

`# -*- coding: utf-8 -*-`

It is written in NVDA's [codingStandards.md](https://github.com/nvaccess/nvda/blob/master/devDocs/codingStandards.md), in the 'Encoding' paragraph.

Even if it is not required for Python 3, it seems it is required in some situations for gettext. See [here](https://nvda-addons.groups.io/g/nvda-addons/topic/84411196#16341).

**Comment by cary-rowen** [comment]
Thanks @CyrilleB79

I define the coding on the first line, but when I run scons pot it still throws an error:
```
cd source && D:\git\nvda\miscDeps\tools\xgettext.exe -o D:\git\nvda\output\nvda_snapshot_source-CorrectedChineseBrailleT
ableNames-4aa8735.pot --package-name NVDA --package-version source-CorrectedChineseBrailleTableNames-4aa8735 --foreign-u
ser --add-comments=Translators: --keyword=pgettext:1c,2 --from-code utf-8 --language=python --files-from=D:\git\nvda\out
put\potSourceFileList.txt
scons: *** [output\nvda_snapshot_source-CorrectedChineseBrailleTableNames-4aa8735.pot] UnicodeDecodeError : 'gbk' codec
can't decode byte 0xad in position 2154: illegal multibyte sequence
Traceback (most recent call last):
  File "D:\git\nvda\.venv\lib\site-packages\SCons\Action.py", line 1279, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "D:\git\nvda\SConstruct", line 474, in makePot
    for lineNum, line in enumerate(inp):
UnicodeDecodeError: 'gbk' codec can't decode byte 0xad in position 2154: illegal multibyte sequence
scons: building terminated because of errors.
Deactivating NVDA Python virtual environment
```

**Comment by CyrilleB79** [comment]
> Thanks @CyrilleB79
> 
> I define the coding on the first line, but when I run scons pot it still throws an error:
> 
> ```
> cd source && D:\git\nvda\miscDeps\tools\xgettext.exe -o D:\git\nvda\output\nvda_snapshot_source-CorrectedChineseBrailleT
> ableNames-4aa8735.pot --package-name NVDA --package-version source-CorrectedChineseBrailleTableNames-4aa8735 --foreign-u
> ser --add-comments=Translators: --keyword=pgettext:1c,2 --from-code utf-8 --language=python --files-from=D:\git\nvda\out
> put\potSourceFileList.txt
> scons: *** [output\nvda_snapshot_source-CorrectedChineseBrailleTableNames-4aa8735.pot] UnicodeDecodeError : 'gbk' codec
> can't decode byte 0xad in position 2154: illegal multibyte sequence
> Traceback (most recent call last):
>   File "D:\git\nvda\.venv\lib\site-packages\SCons\Action.py", line 1279, in execute
>     result = self.execfunction(target=target, source=rsources, env=env)
>   File "D:\git\nvda\SConstruct", line 474, in makePot
>     for lineNum, line in enumerate(inp):
> UnicodeDecodeError: 'gbk' codec can't decode byte 0xad in position 2154: illegal multibyte sequence
> scons: building terminated because of errors.
> Deactivating NVDA Python virtual environment
> ```

Sorry, I do not have any other idea on the issue here.

**Comment by cary-rowen** [comment]
Hi @CyrilleB79
I think it makes sense to use the names suggested in the Chinese national standard documents in NVDA.
Also, I'm a member of the Chinese translation team, and the note to the translator doesn't seem necessary because I'm well aware of what this change represents.
Seriously, as a Chinese translator, I'm confused by the original display name using "Grade1","Grade2" to distinguish the two tables, because no one in China knows such a name.
The NVDA Chinese team also intends to submit a new Chinese Braille table to Liblouis. Obviously this new Braille table should not be named "Grade3".
Anyway, thanks for your help.

**Comment by josephsl** [comment]
Hi,

I'm raising a possible red flag here: even if native speakers know about what the original string and the translated text says, you should still provide translations for benefit of non-native speakers or folks who may not know about national standards. Therefore, I strongly urge you to add translator comments.

A possible compromise would be something I did when adding the 2006 edition of Korean braille to Libluis and NVDA: add the names to be used in China and elsewhere is parentheses. If not, you should still provide translator comments, as without that, this pull request will not proceed.

Thanks.

**Comment by cary-rowen** [comment]
Hi, @josephsl 
Since the translation contains Chinese strings, I defined utf-8 encoding at the beginning of the file, but the translation check still throws an error, and there is no way to go to the next step.
I might need some help, do you have any possible hints?

Thanks

**Comment by josephsl** [comment]
Hi, the translation check is performed not on nvda.po files, but rather on Python files you modify. Any message meant to be shown to user and to be translated must have translator comments specified in the Python file, not nvda.po. Thanks.

**Comment by cary-rowen** [comment]
Hi, I did add translation comments in brailleTables.py.

**Comment by CyrilleB79** [comment]
> Hi, I did add translation comments in brailleTables.py.

@josephsl, to clarify things regarding the technical encoding issue:
The issue is the following:
In the source file `brailleTables.py`, @cary-rowen has added non-Ascii characters in one of the translator comment. The file is correctly encoded in UTF8 (without BOM) and the first line is the encoding specification as defined at Python 2 time. Though, the Translation comments check fails.
Could you help here?

@cary-rowen:
Regarding the translation of braille table names, I understand that grade 1/2 does not make sense for Chinese people. I will let people more familiar with Braille in various languages comment here, in case they have an idea of translator comment that could help clarifying things for non-Chinese translators.

File: source/brailleTables.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+# -*- coding:utf-8 -*-
```
Comment (lukaszgo1) [inline source/brailleTables.py:1]: This encoding declaration is invalid. This should be:
```suggestion
# -*- coding:utf-8 -*-
# -*- coding: UTF-8 -*-
```

**Comment by lukaszgo1** [comment]
As to why the checkPot fails it is because Python `open` function opens the text files using the default ANSI code page on Windows and the ANSI code page on your system (and on AppVeyor) cannot correctly decode utf-8 text.
To fix this you need to specify explicit encoding for the `open` calls in sconstruct i.e. replace:
`with open(potFn, "rt") as inp, open(tmpFn, "wt") as out:`
with:
`with open(potFn, "rt", encoding="utf-8") as inp, open(tmpFn, "wt", encoding="utf-8") as out:`
and do the same for the `open` call in the `checkPot` file in the tests directory.

**Comment by cary-rowen** [comment]
> As to why the checkPot fails it is because Python function opens the text files using the default ANSI code page on Windows and the ANSI code page on your system (and on AppVeyor) cannot correctly decode utf-8 text.
Hi, @lukaszgo1 

Thank you very much for your help, I followed your instructions, made these changes and it works perfectly.
However, this change seems to be outside the scope of this PR, do I need to open a separate PR to fix this?

Thanks Again

**Comment by lukaszgo1** [comment]
@cary-rowen  I'm not sure if this deserves a separate PR. @seanbudd  What do you think?

**Comment by seanbudd** [comment]
> As to why the checkPot fails it is because Python `open` function opens the text files using the default ANSI code page on Windows and the ANSI code page on your system (and on AppVeyor) cannot correctly decode utf-8 text. To fix this you need to specify explicit encoding for the `open` calls in sconstruct i.e. replace: `with open(potFn, "rt") as inp, open(tmpFn, "wt") as out:` with: `with open(potFn, "rt", encoding="utf-8") as inp, open(tmpFn, "wt", encoding="utf-8") as out:` and do the same for the `open` call in the `checkPot` file in the tests directory.

While this is a small change, it is a separate concern and would be better fixed in a separate PR.

File: source/brailleTables.py (hunk 3 above)
Comment (seanbudd) [reply source/brailleTables.py:1]: @cary-rowen did you see this comment?

Comment (seanbudd) [reply source/brailleTables.py:1]: @cary-rowen did you see this comment?

**Comment by cary-rowen** [comment]
Hi, @seanbudd 
I applied the changes suggested by @lukaszgo1 and created #13698 
Can you take a look?

**Review by seanbudd — APPROVED** [review]
Thanks @cary-rowen.

Leaving out the changelog item as it just a minor string change

**Pull request merged**